harness/determined/_core/_searcher.py

```diff
@@ -28,6 +28,20 @@ def _parse_searcher_units(experiment_config: dict) -> Optional[Unit]:
 
 
 class SearcherOperation:
+    """
+    A SearcherOperation is a request from the hyperparameter-search logic for the training script
+    to execute one train-validate-report cycle.
+
+    Some searchers (like single, random, or grid) will pass only a single SearcherOperation to each
+    trial, while others may pass many SearcherOperations.
+
+    Each SearcherOperation has a length attribute representing the cumulative training that should
+    be completed before the validate-report steps of the cycle.  The length attribute is absoute,
+    not incremental, meaning that if the searcher wants you to train for 10 units and validate, then
+    train for 10 more units and validate, it will emit one SearcherOperation with .length=10
+    followed by a second SearcherOperation with .length=20.  Using absolute lengths instead of
+    incremental lengths makes restarting after crashes simple and robust.
```

Comment:
I actually think this is correct.  The subject of the sentence is the gerund "using".

Reply:
Grammarly disagrees but they're not always right and I'm not sure.

Reply:
If you take out the `instead of incremental lengths`, it becomes `Using absolute lengths makes restarting after crashes simple and robust`.  This is the same pattern as "coding makes me happy" vs "coding make me happy"